### DISCUSSION TITLE
Go API: Replace cue.Value instead of Unify

### DISCUSSION THREAD

**Discussion by tombrk** [cue-lang/cue#1200] [General]
Hey all!

I have the following use-case:

- A _single_ local `.cue` (e.g. `foo.cue`) file is evaluated to a `cue.Value` using the Go API
- Some nested value (e.g. `bar.baz`) is serialized to JSON and presented to the user
- The user may perform edits on that JSON
- The Go application receives the JSON back, which now possibly no longer unifies, given it was edited

Is there a way to replace whatever was at `bar.baz` originally in the root `cue.Value` with what we received as JSON?

**Comment by myitcv** [comment]
From your description I think you are looking for https://pkg.go.dev/cuelang.org/go@v0.4.0/cue#Value.FillPath. But if I have misunderstood, please can you provide a concrete example?

**Reply by verdverm** [reply]
My read is that OP wants to change a concrete value within a larger value. Can FillPath change an existing concrete field?

This line makes me believe it cannot.

https://github.com/cue-lang/cue/blob/3b0a537a2db6d49d1ae67631a31c45ad97c69078/cue/types.go#L1712

A reproducer would make this more clear, once at a full keyboard

**Reply by myitcv** [reply]
> Can FillPath change an existing concrete field?

No it can't, which is why a concrete example is going to help steer this. Because I suspect there needs to be a separation into at least two CUE values here: the template (or schema) and the default value (which is passed as JSON to the user). That way, the modified result from the user can be decoded into a `cue.Value` and simply unified with the template (schema) to verify correctness. Of course if that check step is not required, simply decode a `cue.Value` from the JSON.

**Reply by verdverm** [reply]
The intersection with Fill and the Query proposal could be of interest here. If I have a large schema and concrete value, and then want to change a single field, the logic to fill the new value becomes simpler if there were a FillQuery
